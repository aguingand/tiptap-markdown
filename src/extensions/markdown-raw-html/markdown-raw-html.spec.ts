import { expect, test } from "vitest";
import { parse } from "../../../__tests__/utils";

test('parse block html node', () => {
    expect(parse('<custom-element>\nexample\n</custom-element>', {
        htmlNode: {
            group: 'block',
            content: 'inline*',
            parseHTML: () => [{
                tag: 'custom-element',
            }],
        },
    })).toMatchSnapshot();
});
test('parse block html node inner <p>', () => {
    expect(parse('<custom-element>\n\nexample\n\n</custom-element>', {
        htmlNode: {
            group: 'block',
            content: 'block*',
            parseHTML: () => [{
                tag: 'custom-element',
            }],
        },
    })).toMatchSnapshot();
});
test('parse inline html node', () => {
    expect(parse('<custom-element></custom-element>', {
        htmlNode: {
            group: 'inline',
            inline: true,
            content: 'text*',
            parseHTML: () => [{
                tag: 'custom-element',
            }],
        },
    })).toMatchSnapshot();
});
test('parse inline html node inner text', () => {
    expect(parse('<custom-element>example</custom-element>', {
        htmlNode: {
            group: 'inline',
            inline: true,
            content: 'text*',
            parseHTML: () => [{
                tag: 'custom-element',
            }],
        },
    })).toMatchSnapshot();
});
test('parse block disabled', () => {
    expect(parse('<custom-element>\nexample\n</custom-element>', {
        markdown: {
            html: false,
        },
    })).toBeNull();
});
test('parse block disabled inner <p>', () => {
    expect(parse('<custom-element>\n\nexample\n\n</custom-element>', {
        markdown: {
            html: false,
        },
    })).toMatchSnapshot();
});
test('parse inline disabled', () => {
    expect(parse('<custom-element></custom-element>', {
        markdown: {
            html: false,
        },
    })).toMatchSnapshot();
});

test('parse inline disabled inner text', () => {
    expect(parse('<custom-element>example</custom-element>', {
        markdown: {
            html: false,
        },
    })).toMatchSnapshot();
});

