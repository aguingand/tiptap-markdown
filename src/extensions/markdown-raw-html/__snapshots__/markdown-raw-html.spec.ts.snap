// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`parse block disabled inner <p> 1`] = `
[
  {
    "content": [
      {
        "text": "example",
        "type": "text",
      },
    ],
    "type": "paragraph",
  },
]
`;

exports[`parse block html node 1`] = `
[
  {
    "content": [
      {
        "text": "example",
        "type": "text",
      },
    ],
    "type": "html-node",
  },
]
`;

exports[`parse block html node inner <p> 1`] = `
[
  {
    "content": [
      {
        "content": [
          {
            "text": "example",
            "type": "text",
          },
        ],
        "type": "paragraph",
      },
    ],
    "type": "html-node",
  },
]
`;

exports[`parse inline disabled 1`] = `
[
  {
    "type": "paragraph",
  },
]
`;

exports[`parse inline disabled inner text 1`] = `
[
  {
    "content": [
      {
        "text": "example",
        "type": "text",
      },
    ],
    "type": "paragraph",
  },
]
`;

exports[`parse inline html node 1`] = `
[
  {
    "content": [
      {
        "type": "html-node",
      },
    ],
    "type": "paragraph",
  },
]
`;

exports[`parse inline html node inner text 1`] = `
[
  {
    "content": [
      {
        "content": [
          {
            "text": "example",
            "type": "text",
          },
        ],
        "type": "html-node",
      },
    ],
    "type": "paragraph",
  },
]
`;
